qcodes/dataset/measurements.py

```diff
@@ -90,31 +103,56 @@ def add_result(self,
         are not unraveled but stored directly for improved performance.
 
         Args:
-            res: a dictionary with keys that are parameter names and items
-                that are the corresponding values at this measurement point.
+            res_tuple: a tuple with the first element being the parameter name
+                and the second element is the corresponding value(s) at this
+                measurement point. The function takes as many tuples as there
+                are results.
 
         Raises:
             ValueError: if a parameter name not registered in the parent
                 Measurement object is encountered.
             ParameterTypeError: if a parameter is given a value not matching
                 its type.
         """
-        res = list(res_tuple)  # ArrayParameters cause us to mutate the results
+        # ArrayParameters and MultiParameters cause us to mutate
+        # the results.
```

Comment:
is this comment still relevant? you don't seem to mutate results, instead a new "list of resutls" is created now, right? also, does this change impact performance?

Reply:
Removed the comment. 
My preliminary benchmarks indicate that the difference is small as long as the number of tuples added is small. I therefore think this is worth the performance trade off. The alternative would involve removing paramter tuples from the list for multiparameters. That does not seem like a good idea